canarytokens/aws_infra/plan_generation.py

```diff
@@ -311,44 +351,145 @@ def _get_ingestion_bus_arn(bus_name: str):
+        AssetLabel.SSM_PARAMETER_NAME,
+        AssetLabel.SECRET_NAME,
+        AssetLabel.TABLE_NAME,
+    }
+
+    # Child asset types (nested resources)
+    child_fields = {
+        AssetLabel.OBJECT_PATH,
+        AssetLabel.TABLE_ITEM,
+    }
+
+    if asset_field in parent_fields:
+        return await _generate_parent_asset_name(asset_type, inventory)
+    elif asset_field in child_fields:
+        return await _generate_child_asset_name(asset_type, parent_asset_name)
+    else:
+        raise ValueError(f"Unsupported asset field: {asset_field}")
+
+
+async def generate_data_choice(
+    canarydrop: Canarydrop,
+    asset_type: AWSInfraAssetType,
+    asset_field: str,
+    parent_asset_name: str = None,
+) -> str:
+    """Generate a random data choice for the given asset type and field."""
+    inventory = _get_inventory_for_asset_type(canarydrop, asset_type)
+
+    return await _generate_asset_by_type_and_field(
+        asset_type, asset_field, inventory, parent_asset_name
+    )
+
+
+async def generate_child_assets(assets: dict):
     """
-    Generate a random data choice for the given asset type and field.
+    Generate child assets for the given assets.
     """
-    asset_map = {
-        (AWSInfraAssetType.S3_BUCKET, "bucket_name"): generate_s3_bucket,
-        (AWSInfraAssetType.S3_BUCKET, "object_path"): generate_s3_object,
-        (AWSInfraAssetType.SQS_QUEUE, "queue_name"): generate_sqs_queue,
-        (AWSInfraAssetType.SQS_QUEUE, "message_count"): lambda: str(
-            random.randint(0, 10)
-        ),
-        (AWSInfraAssetType.SSM_PARAMETER, "parameter_name"): generate_ssm_parameter,
-        (
-            AWSInfraAssetType.SECRETS_MANAGER_SECRET,
-            "secretsmanager_secret_name",
-        ): generate_secretsmanager_secret,
-        (
-            AWSInfraAssetType.SECRETS_MANAGER_SECRET,
-            "secretsmanager_secret_value",
-        ): lambda: "".join(
-            random.choice(string.ascii_letters + string.digits)
-            for _ in range(random.randint(5, 50))
-        ),
-        (AWSInfraAssetType.DYNAMO_DB_TABLE, "dynamodb_name"): generate_dynamo_table,
-        (AWSInfraAssetType.DYNAMO_DB_TABLE, "dynamodb_partition_key"): lambda: "".join(
-            random.choice(string.ascii_letters + string.digits)
-            for _ in range(random.randint(3, 10))
-        ),
-        (AWSInfraAssetType.DYNAMO_DB_TABLE, "dynamodb_row_count"): lambda: str(
-            random.randint(0, 100)
-        ),
+    result = {
+        AWSInfraAssetType.S3_BUCKET.value: {},
+        AWSInfraAssetType.DYNAMO_DB_TABLE.value: {},
     }
+    tasks = []
+    for asset_type, asset_names in assets.items():
+        for asset_name in asset_names:
+            tasks.append(
+                NAME_GENERATOR.generate_children_names(
+                    asset_type, asset_name, random.randint(1, MAX_CHILD_ITEMS)
+                )
+            )
+    all_names = await asyncio.gather(*tasks)
```

Comment:
Is this a `list[list[str]]`? Does that mean we're assigning a list of names to each `result[asset_type][asset_name]` below?

Trying to wrap my head around the return type here... it's not immediately obvious. 
It might even be worth making it explicit too, with something like:
```suggestion
    all_names: list[list[str]] = await asyncio.gather(*tasks)
```
Or consider renaming it to something like `name_lists`.


If it's a `list[str]`, then I'd suggest the annotation
```suggestion
    all_names: list[str] = await asyncio.gather(*tasks)
```